### PR TITLE
Beginning C++23: Chapter 20

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

File: Examples/Modules/Chapter 20/Ex20_12A/Ex20_12A.cpp [hunk 1]
```diff
@@ -1,30 +1,27 @@
...
+import std;
+import box;      // From Ex13_03
 
 int main()
 {
   std::vector boxes{ Box{1,2,3}, Box{5,2,3}, Box{9,2,1}, Box{3,2,1} };
 
   // Define a lambda functor to print the result of find() or find_if():
-  auto print_result{ [&boxes](auto result)
+  auto print_result = [&boxes] (auto result)
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 20/Ex20_12A/Ex20_12A.cpp:10]: Why no longer uniform init?

File: Examples/Modules/Chapter 20/Ex20_19/Ex20_19.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// Converting ranges to containers
+import std;
+
+bool isOdd(int i) { return i % 2 == 1; }
+
+int main()
+{
+  auto view = std::views::iota(1, 11); // Could be any view/range...
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 20/Ex20_19/Ex20_19.cpp:8]: There are lots of places in this function where you can use uniform init.

File: Examples/Modules/Chapter 20/Ex20_19/Ex20_19.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  // Same operations, but now using pre-C++23 iterator-pair based functions:
+  auto odds = set | std::views::filter(isOdd);
+  std::vector<int> v2{ odds.begin(), odds.end() };
+  
+  std::vector<int> v3;
+  const auto atoi = view | std::views::reverse;
+  v3.assign(atoi.begin(), atoi.end());
+  const auto repeated = std::views::repeat(-1, 3);
+  v3.insert(v3.end(), repeated.begin(), repeated.end());  // No append()!
+  v3.insert(v3.begin(), set.begin(), set.end());
```
Comment (MarcGregoire) [inline Examples/Modules/Chapter 20/Ex20_19/Ex20_19.cpp:34]: "v3.begin() + 5"
to match the v1.begin() + 5 above.

File: Exercises/Modules/Chapter 20/Soln20_15/Soln20_15.cpp [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// Implementing a custom std::formatter<> specialization to format Box objects
+import box;
+import box.formatter;
+import std;
+
+int main()
+{
+  Box box{ 1, 2, 3 };
+  std::cout << std::format("My new box, {:.2}, is fabulous!", box);
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 20/Soln20_15/Soln20_15.cpp:9]: Why not:
`std::println("My new box, {:.2}, is fabulous!", box);`

File: Exercises/Modules/Chapter 20/Soln20_16/BoxFormatter.cppm [hunk 5]
```diff
@@ -0,0 +1,50 @@
+export module box.formatter;
+
+import box;
+import std;
+
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 20/Soln20_16/BoxFormatter.cppm:5]: Use Uniform init in this code?

File: Exercises/Modules/Chapter 20/Soln20_17/BoxFormatter.cppm [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    auto closing_brace = std::ranges::find(context, '}');
+
+    if (closing_brace == context.end()) // If no '}' found, fail
+      throw std::format_error{ "missing closing braces, }" };
+
+    m_no_box = (closing_brace != context.begin() && *(closing_brace - 1) == 'n');
+
+    std::string double_format(context.begin(), closing_brace);
+    if (m_no_box)
+      double_format.erase(double_format.end() - 1); // Take out the 'n'
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 20/Soln20_17/BoxFormatter.cppm:25]: Or:
`double_format.pop_back();`

File: Exercises/Modules/Chapter 20/Soln20_17/BoxFormatter.cppm [hunk 7]
```diff
@@ -0,0 +1,56 @@
+export module box.formatter;
+
+import box;
+import std;
+
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 20/Soln20_17/BoxFormatter.cppm:5]: Use Uniform init in this code?

File: Exercises/Modules/Chapter 20/Soln20_17/BoxFormatter.cppm [hunk 8]
```diff
@@ -0,0 +1,56 @@
+export module box.formatter;
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 20/Soln20_17/BoxFormatter.cppm:1]: Very nice new exercises about custom formatters!

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

File: Examples/Modules/Chapter 20/Ex20_05/Ex20_05.cpp [hunk 9]
```diff
@@ -1,33 +1,28 @@
...
 {
   std::string text;    // The string to count words in
 
   // Read a string from the keyboard
-  std::cout << "Enter a string terminated by *:" << std::endl;
+  std::println("Enter a string terminated by *:");
   getline(std::cin, text, '*');
 
-  const Words words{ extractWords(text) };
+  Words words = extractWords(text);
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 20/Ex20_05/Ex20_05.cpp:22]: I'll never get used to it... Good thing that it also makes no difference 99% of the times, and the times where it does make a difference it's 99% an irrelevant one :-)

File: Examples/Modules/Chapter 20/Ex20_12A/Ex20_12A.cpp (hunk 1 above)
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 20/Ex20_12A/Ex20_12A.cpp:10]: But... but... It's so ugly for lambda expressions...

File: Exercises/Modules/Chapter 20/Soln20_15/Soln20_15.cpp (hunk 4 above)
Comment (Peter-VanWeert) [reply Exercises/Modules/Chapter 20/Soln20_15/Soln20_15.cpp:9]: Otherwise you wouldn't have any work...

**Pull request merged**